src/components/NextPrevious/index.tsx

```diff
@@ -4,22 +4,26 @@ import directory from '@/directory/directory.json';
 import { useRouter } from 'next/router';
 import { useCurrentPlatform } from '@/utils/useCurrentPlatform';
 import { IconChevron } from '@/components/Icons';
+import { useIsGen1Page } from '@/utils/useIsGen1Page';
 
 export const NextPrevious = () => {
   const platform = useCurrentPlatform();
   const router = useRouter();
   const pathname = router.pathname;
+  const isGen1 = useIsGen1Page();
 
-  const findDirectoryNodes = (
-    route,
-    dir = directory,
-    platform,
-    previous,
-    next
-  ) => {
+  const findDirectoryNodes = (route, dir, platform, previous, next) => {
     const children = dir?.children?.filter((child) => {
-      return child?.platforms?.includes(platform);
+      if (isGen1) {
+        return (
+          child?.route.startsWith('/gen1') &&
+          child?.platforms?.includes(platform)
+        );
+      } else {
+        return child?.platforms?.includes(platform);
+      }
```

Comment:
Needed to update the logic here so that we search the directory object correctly. Previously, we filter a node's children based if it had the right platform. Now we also need to look to see if we're on a gen1 page or not